cognite/client/config.py

```diff
@@ -163,3 +200,45 @@ def default(
             credentials=credentials,
             base_url=f"https://{cdf_cluster}.cognitedata.com/",
         )
+
+    @classmethod
+    def load(cls, config: dict | str) -> ClientConfig:
+        """Loads a dictionary of configuration fields into a client config object.
+
+        Args:
+            config (dict | str): A dictionary or dictionary parsable string containing configuration values defined in the ClientConfig class.
+
+        Returns:
+            ClientConfig: A client config object.
+
+        Examples:
+                Create a client config object from a dictionary input:
+
+                >>> from cognite.client.config import ClientConfig
+                >>> import os
+                >>> config = {
+                ...     "client_name": "abcd",
+                ...     "project": "cdf-project",
+                ...     "base_url": "https://api.cognitedata.com/",
+                ...     "credentials": {
+                ...         "client_credentials": {
+                ...             "client_id": "abcd",
+                ...             "client_secret": os.environ["OAUTH_CLIENT_SECRET"],
+                ...             "token_url": os.environ["TOKEN_URL"],
+                ...             "scopes": ["https://greenfield.cognitedata.com/.default"],
+                ...             # Any additional IDP-specific token args. e.g.
+                ...             "audience": "some-audience",
+                ...         },
+                ...     },
+                ... }
+                >>> client_config = ClientConfig.load(config)
+        """
+        loaded = load_dict_or_str(config)
+
+        try:
+            credentials_config_input = loaded.pop("credentials")
+        except KeyError:
+            raise ValueError("'credentials' is a required field and must be included in the input dictionary.")
+
+        credentials = CredentialProvider.load(credentials_config_input)
+        return ClientConfig(credentials=credentials, **loaded)
```

Comment:
(TL;DR) Let's write out all the arguments explicitly

Reply:
I'm not manipulating any of the other args though, so don't we just want to leave it to the `__init__` to use the unpacked dict?

Reply:
Yeah, we've had enough issues with old SDK versions and dict unpacking in the past... 😅 
```py
return ClientConfig(
    client_name=config["client_name"],
    project=config["project"],
    credentials=config["credentials"],
    api_subversion=config.get("api_subversion"),
    base_url=config.get("base_url"),
    max_workers=config.get("max_workers"),
    headers=config.get("headers"),
    timeout=config.get("timeout"),
    file_transfer_timeout=config.get("file_transfer_timeout"),
    debug=config.get("debug"),
)

Reply:
Ah ok, then I'll do that for all of them no problem 👌🏻